### PR TITLE
Untag memory used for stack

### PR DESCRIPTION
**Pull request by fmayer**
Old versions (including the one currently in use in Android) of the HWASan runtime do not support a tagged stack pointer.

Tested by building and running the unittests on aarch64 with

  export CFLAGS='-fsanitize=hwaddress -mllvm -hwasan-globals=0'
  export CXXFLAGS='-fsanitize=hwaddress -mllvm -hwasan-globals=0'

Globals don't currently work with glibc, so I disabled them.

Verified without this change the unittests crash with a HWASan error.

### CODE REVIEW

File: src/osfiber_aarch64.c [hunk 1]
```diff
@@ -23,12 +25,23 @@ void marl_fiber_trampoline(void (*target)(void*), void* arg) {
...
+__attribute((weak)) void *__hwasan_tag_pointer(const volatile void *p,
+    unsigned char tag);
+
 MARL_EXPORT
 void marl_fiber_set_target(struct marl_fiber_context* ctx,
                            void* stack,
                            uint32_t stack_size,
                            void (*target)(void*),
                            void* arg) {
+  if (__hwasan_tag_memory && __hwasan_tag_pointer) {
```
Comment (ben-clayton) [inline src/osfiber_aarch64.c:41]: Is there a preprocessor define we can use to guard this?

**Review by ben-clayton — COMMENTED** [review]
Thanks! One small question

File: src/osfiber_aarch64.c (hunk 1 above)
Comment (fmayer) [reply src/osfiber_aarch64.c:41]: Not really, on Android it is possible to run an instrumented libc (which returns tagged pointers) even if your program is compiled as normal, so we won't be able to know whether we are running with HWASan at compile time.

Comment (ben-clayton) [reply src/osfiber_aarch64.c:41]: Fair enough.

**Pull request merged**